### PR TITLE
Creates a servlet that stores a scheduledInterview in Datastore

### PR DESCRIPTION
**Pull request by GrantComm**
Tested using hard-coded request in JavaScript (changes to JS files will go in separate pr)

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:27]: Need to remove these blank lines.
https://google.github.io/styleguide/javaguide.html#s3.3-import-statements

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    this.scheduledInterviewDao = scheduledInterviewDao;
+  }
+
+  // Sends the request's contents to Datastore in the form of a new ScheduledInterview.
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy");
+    ScheduledInterview scheduledInterview =
+        ScheduledInterview.create(
+            0,
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java]: Nit: l think Jack suggested -1, because 0 could still be a legit id in Datastore.

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  // TODO: add a FakeScheduledInterviewDao class so this will become useful
+  public void init(ScheduledInterviewDao scheduledInterviewDao) {
+    this.scheduledInterviewDao = scheduledInterviewDao;
+  }
+
+  // Sends the request's contents to Datastore in the form of a new ScheduledInterview.
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy");
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java]: When is this ever used?

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    ScheduledInterview scheduledInterview =
+        ScheduledInterview.create(
+            0,
+            new TimeRange(
+                Instant.parse(request.getParameter("startTime")),
+                Instant.parse(request.getParameter("endTime"))),
+            LocalDate.parse(request.getParameter("date")),
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java]: Don't need date in Datastore, but that should be in another PR.

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  // Sends the request's contents to Datastore in the form of a new ScheduledInterview.
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy");
+    ScheduledInterview scheduledInterview =
+        ScheduledInterview.create(
+            0,
+            new TimeRange(
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:49]: Datastore can't represent TimeRange. What l'm doing with Availability is converting TimeRange's lnstants to longs using https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#toEpochMilli--

and storing longs in Datastore. 
l'm thinking on the front end we should have these longs stored in some data- attribute in the HTML and we just pass them in as longs here.
lt might honestly be more worth it to get some of the front end set up first, so we know what's being passed in exactly.

**Review by apluscs — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java (hunk 5 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:49]: I'm currently storing the instants as strings in Datastore via the create method in DatastoreScheduledInterview.java but I can change them to longs instead

Comment (apluscs) [reply src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:49]: Hm yeah it'd be better if it matched with Availability entities

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.time.Instant;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/new-scheduled-interview")
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:28]: This is going to be used eventually to get a user's scheduled interviews as well right? So remove the `new-`

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+public class NewScheduledInterviewServlet extends HttpServlet {
```
Comment (apluscs) [inline src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:29]: New here is questionable too

**Review by apluscs — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java (hunk 6 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/servlets/NewScheduledInterviewServlet.java:28]: I have them as two separate files, there's another servlet for listing scheduledInterviews
